configuration.k

```diff
@@ -0,0 +1,228 @@
+
+module CONFIGURATION
+
+imports SOLIDITY-SYNTAX
+
+
+syntax START
+
+
+configuration
+    <k> $PGM:START </k>
+    <exit-code exit=""> 1 </exit-code>
+    // ... 
+    
+    <ethereum>
+        
+        // Solidity execution engine
+        // =========================
+        
+        <solidity> 
+            
+            // Mutable during a single transaction
+            // -----------------------------------
+            
+            <execEngine>
+                <output>          .K  </output>
+                <statusCode>      .K  </statusCode>
+                <callStack>     .List  </callStack>
+                <interimStates>   .K  </interimStates>
+                <touchedAccounts> .K  </touchedAccounts>
+                <currentAccount>  #account(#undef_Int,#undef_Id)  </currentAccount>
```

Comment:
This is the only place where `#undef_Int` and `#undef_Id` are used. I would recommend instead creating a value `.Account` or `#undef_Account` like so:

```
syntax AccountId ::= Int | "#undef_Account"
```

You declare `#undef_Int` for example here:

```
syntax Int ::= "#undef_Int"
```

which means you're adding junk to the `Int` sort, which will make reasoning using unification and Z3 very difficult when it comes time to do the proofs. Almost always better to use a super-sort (like the `AccountId` super-sort above) and add the extra constants there.

Reply:
@shangweilin is it ever the case that it's useful to have an `#account` with only one of the entries undefined? Eg: `#account(#undef_Int, x)` or `#account(3, #undef_Id)`? If not (and I can't see any uses in the semantics of these cases), then instead of adding sorts `AccountId` and `AccountInt` as you have, I would just add an extra constant `#undef_Account` to sort `Account`.

```
syntax Account ::= #account(Int, Id) | "#undef_Account"
```

If the semantics ever could produce examples like above, then it would make sense, but otherwise would be better to avoid adding super-sorts of builtin sorts (like `Int` and `Id`) if possible.

Note that this pattern of adding an extra constant to a sort (like `#undef_Account`) is like the `Maybe` type in Haskell, or the `Optional` in Java/Scala.